internal/tools/aws/database_multitenant_pgbouncer_test.go

```diff
@@ -0,0 +1,329 @@
+}
+
+func TestNewRDSMultitenantPGBouncerDatabase_DefaultLimit(t *testing.T) {
+	database := NewRDSMultitenantPGBouncerDatabase(
+		model.DatabaseEngineTypePostgresProxy,
+		"instance-id",
+		"installation-id",
+		&Client{},
+		0, // Should use default
+		false,
+	)
+
+	assert.Equal(t, DefaultRDSMultitenantPGBouncerDatabasePostgresCountLimit, database.maxSupportedInstallations)
+}
+
+func TestRDSMultitenantPGBouncerDatabase_IsValid(t *testing.T) {
+	tests := []struct {
+		name           string
+		databaseType   string
+		installationID string
+		expectError    bool
+		errorMessage   string
+	}{
+		{
+			name:           "valid configuration",
+			databaseType:   model.DatabaseEngineTypePostgresProxy,
+			installationID: "test-installation-id",
+			expectError:    false,
+		},
+		{
+			name:           "empty installation ID",
+			databaseType:   model.DatabaseEngineTypePostgresProxy,
+			installationID: "",
+			expectError:    true,
+			errorMessage:   "installation ID is not set",
+		},
+		{
+			name:           "invalid database type",
+			databaseType:   "invalid-type",
+			installationID: "test-installation-id",
+			expectError:    true,
+			errorMessage:   "invalid pgbouncer database type invalid-type",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			database := &RDSMultitenantPGBouncerDatabase{
+				databaseType:   tt.databaseType,
+				installationID: tt.installationID,
+			}
+
+			err := database.IsValid()
+
+			if tt.expectError {
+				assert.Error(t, err)
+				assert.Contains(t, err.Error(), tt.errorMessage)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestRDSMultitenantPGBouncerDatabase_DatabaseEngineTypeTagValue(t *testing.T) {
+	database := &RDSMultitenantPGBouncerDatabase{}
+	tagValue := database.DatabaseEngineTypeTagValue()
+	assert.Equal(t, DatabaseTypePostgresSQLAurora, tagValue)
+}
+
+func TestRDSMultitenantPGBouncerDatabase_MaxSupportedDatabases(t *testing.T) {
+	database := &RDSMultitenantPGBouncerDatabase{
+		maxSupportedInstallations: 150,
+	}
+	maxDatabases := database.MaxSupportedDatabases()
+	assert.Equal(t, 150, maxDatabases)
+}
+
+// Test SQL query generation for PGBouncer operations
+func TestPGBouncerSQLQueries(t *testing.T) {
+	t.Run("insert user query format", func(t *testing.T) {
+		username := "test_user"
+		scramHash := "SCRAM-SHA-256$4096:c2FsdA==$stored:server"
+
+		expectedQuery := fmt.Sprintf(`INSERT INTO pgbouncer.pgbouncer_users (usename, passwd) VALUES ('%s', '%s')`, username, scramHash)
+
+		assert.Contains(t, expectedQuery, "INSERT INTO pgbouncer.pgbouncer_users")
+		assert.Contains(t, expectedQuery, username)
+		assert.Contains(t, expectedQuery, scramHash)
+		assert.Contains(t, expectedQuery, "usename")
+		assert.Contains(t, expectedQuery, "passwd")
+
+		// Ensure proper SQL formatting
+		assert.True(t, strings.Count(expectedQuery, "'") >= 4) // At least 4 single quotes for string values
+	})
```

Comment:
Quite a few of these tests really aren't covering anything in the codebase. This one for example just creates a string and then checks if certain substrings exist in it. This could be changed into something more useful if it imported constants or functions from the code, but as is won't really help prevent bugs or regressions. There are others like it below.

Reply:
will remove them in another PR to avoid building the image again